tests/measurements/test_purity_measurement.py

```diff
@@ -0,0 +1,374 @@
+import pytest
+
+import numpy as np
+import pennylane as qml
+
+
+def expected_purity_ising_xx(param):
+    """Returns the analytical purity for subsystems of the IsingXX"""
+
+    eig_1 = (1 + np.sqrt(1 - 4 * np.cos(param / 2) ** 2 * np.sin(param / 2) ** 2)) / 2
+    eig_2 = (1 - np.sqrt(1 - 4 * np.cos(param / 2) ** 2 * np.sin(param / 2) ** 2)) / 2
+    return eig_1**2 + eig_2**2
+
+
+def expected_purity_grad_ising_xx(param):
+    """The analytic gradient purity for the IsingXX"""
+
+    eig_1 = (1 + np.sqrt(1 - 4 * np.cos(param / 2) ** 2 * np.sin(param / 2) ** 2)) / 2
+    eig_2 = (1 - np.sqrt(1 - 4 * np.cos(param / 2) ** 2 * np.sin(param / 2) ** 2)) / 2
+    grad_expected_purity = (
+        2
+        * eig_1
+        * (np.sin(param / 2) ** 3 * np.cos(param / 2) - np.sin(param / 2) * np.cos(param / 2) ** 3)
+        / np.sqrt(1 - 4 * np.sin(param / 2) ** 2 * np.cos(param / 2) ** 2)
+    ) + (
+        2
+        * eig_2
+        * (
+            np.sin(param / 2)
+            * np.cos(param / 2)
+            * (np.cos(param / 2) ** 2 - np.sin(param / 2) ** 2)
+        )
+        / np.sqrt(1 - 4 * np.sin(param / 2) ** 2 * np.cos(param / 2) ** 2)
+    )
+    return grad_expected_purity
+
+
+class TestPurity:
+    """Tests for purity measurements"""
+
+    devices = ["default.qubit", "lightning.qubit", "default.mixed"]
+    grad_supported_devices = ["default.qubit", "default.mixed"]
+    mix_supported_devices = ["default.mixed"]
+
+    diff_methods = ["backprop", "finite-diff"]
+    return_new = [True, False]
+
+    parameters = np.linspace(0, 2 * np.pi, 10)
```

Comment:
Can we decrease the number of `parameters` to 3, for example? Otherwise the tests take too long:
```suggestion
    parameters = np.linspace(0, 2 * np.pi, 3)
```